### ISSUE TITLE
ifdef removed when used on inline keyword in function

### ISSUE THREAD

**Issue by TheAngryByrd** [fsprojects/fantomas#2017]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAdAdliEA2iALmFgMQCWAZmAIQAiAogEICqA4jrl%2BZnj4mUSZ4VTlwgBbAIYAHMNSkAnMAF4wAQQDOAT0zQAdNLkAKAEqJNAVzyFDs%2BQEowSyABpwUTUUI8A5ptQwAG0PCFBxKAASQm0ZZDRIHkI3UNxISIA3KTxLeODUrgAGdwiISABJYSFCAGVyAC9kEtKAFgKwAF1UgF9msPb0mLjAxMxkvrTM7NzAkNKwAEYJrkgAWSkADwAZfi2hX0IACxT5hYAmQvaurl7U8PFB2LzIIXgAeUodzEQamIITh5THJ5OalM7LNIMYQfL5NdplPCUSBXHoTe4raJPEYAI1gsDwAIxWWBs3hYAAzBCypUvIpCAAxHjZAByiAA7nxvoTxIRFLkUTc0QNMcMErj8dyykCZmhQREWlSoDVEBJyHA8LBMOpCFD3p9%2BJLcJRsl4Bbhblx0ZMhs9xQSIeliTL8vMAKyKyA1GRSaCIJiISiwRSIAAKSikEiIiEUhogvP5EWu5qFEUeoqgdsNjumILJADYPV6fX6A0HEFtYGzo9ApF5KhlYDWfJrY2B44gzRALbgrVKbTi8fbhU7c-MAOyF72%2B-2B4MsGRxRQ1uuYBtN8gtxXGvCmxOou7C-tiwdZ6Wj0oADknxZnZYAwnha5o75rNPHoIQg63t7vxEmuymgJHhmJ4OmepLzAAnNe06lsGqwqti0bfiaHZ7oKB6piKtqgcOOYQaUCzFGSnpTiWs6IC%2BGqYChO5oX%2B%2B6WoeWLHhKYEjgREQLEsJFFr66iUIQ0YvhI0itu2nZgN2-RYcBkCZux%2BGymS5wweRZbKqq6qbmSP70Vw-5SYBGJyQpeEksppyUrxZECUJiiaWq%2BI6fMEnocmmFASxIFseZzpyuICwKiRlRIGMbyYAAKoo2gAOrkEctG-gZjE9sx6bybhWEcZZhHujZxbqIosCWMIdCIHwqr2eJfL6bghnSWAvZRHJSSnjlLqEQWJHrBs5SUJFhxCAwdE1IcQaEPF8CJYqLSXO5AGeSZ3mjOMfnnlxE49ZslSUOQGxvIuUifooDAbDIwaaJoG40YqrrzQxGFMbJK1tYpFmdVxV7bRs5hwIo8BTTNZJzZJjXNWmzxvetnGBdBP1-UG8DMpYEhIYoHzlEJEgBIqCxg8Z1orRI1g%2BJyiD0kG0iEPZkVYu9-lkhcHqIwDqyk%2BQ5OU4o1PVR60CHOGH7RgA%2Bmy5DTccBNLUTGXQ9lSmfeIZw8fMaybEVihSNobyKDsb5A8cs0PSlT1pS9ctjO1isBVwZzgj9mva7r%2BuECjaPRpj2O4yp0vPV5GUkzYnP8NzvPRnTwwMxtyvWWrTs63r5Bvuzwdc1Tx18yRAtC-ZYsS4lfvmwHUNW9HsN28Fau9QAajmTA8CImC%2BIbrYXib9WpTJJcjPLgIdbbuBnPl1ebPSpUfjdDfCH4rfG0X3fLZba0Kx9g8QGc3WjxsdCwIQbBEGdF0WNdmpz2S90L016U4b5q%2BM-MZxbdXHPk0wGrQAA1kwWtf0QmgRRqBGSi%2BJUa3V0qhK%2BENsIDjvv3G2TNvpq1ZByfg-pCBViEJHRA5U9qYASjddQwgEIe0UD7eYekoE31gUOe%2BMc7bwzVgAaUQIgGQ-VBpCEqEAyMsIkp1QgA1QmfZXplxhrlCI5JiLbxGqqTQhxz7zFBgtIyMsREZTMnQiuuBySq1KJ6PwBAiq%2BFRtUWKiBsSrAQHCChkCVHg2oaxWh8C15knJA7eOfBfCYHHnoZsmA6jeOOpYYMkVYChWqMdG6-CqEW1vs4okCD5jkjjvo9QNhoyYCiRkcsmpfAkPRrgngBDXwxPscIlqK1NEuIfqUckVd9Gpx8LCLYEZsTwCkA%2BWA11m4oPJmUx6Hl-ZL3idbVxySR76LoMnKQ2ICCyOTocGouhCCbAGabIZxcRk0LGbUyRW99EsLYREsYlQf5SD0EbCBdFYk9ycbs%2BhOjn76Pfhcz%2BsJNBFRKsIVkb5EDwCaSHb4R9Lqn0wOQ0oblBmLWGbLUZ5cJHiHJEgl5ShbzBmmZoaAihyCqiyUJeALB8GanKtAR8Wt-HrM7mbRecKdkIqVlwckjD9G-BxR%2BKxSAqWCJ6B4DozREiaHpNdQIlDuhAA)

#### Code

```fsharp


    let 
#if !DEBUG
        inline
#endif
        map f ar = Async.map (Result.map f) ar
```
            

#### Result

```fsharp
let inline map f ar = Async.map (Result.map f) ar

```
            
#### Problem description

I'm adding AltCover to FsToolkit.ErrorHandling, however there is a [known issue](https://github.com/SteveGilham/altcover/issues/128) with regards to `inline` keyword and getting correct code coverage.  So the suggest fix is to add an `ifdef` around the `inline` keyword.  However, when running code formatting Fantomas removes the change. 

Yes, I hate this workaround for coverage as much as you probably do.

#### Extra information

- [ ] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas 4.6 branch at 1/1/1990

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

**Comment by nojaf** [comment]
Hey Jimmy, thank you for reporting this issue.

The problem here is that we do not consider the `inline` keyword as an anchor point to attach trivia (`#if !DEBUG`) to.
In a perfect world, the range of the `inline` keyword is provided by the compiler, but in our case, we could capture it via the F# tokens.

Adding it to:
https://github.com/fsprojects/fantomas/blob/24705b93140e2e4bc3983cf15c9018a2711b8daf/src/Fantomas/TokenParser.fs#L1182-L1203

And extending https://github.com/fsprojects/fantomas/blob/24705b93140e2e4bc3983cf15c9018a2711b8daf/src/Fantomas/TriviaTypes.fs#L6

Having that would attach the trivia to the correct node.
So when we come across it in CodePrinter, we can print the attached trivia when we encounter the inline keyword.

Somewhere around:

https://github.com/fsprojects/fantomas/blob/24705b93140e2e4bc3983cf15c9018a2711b8daf/src/Fantomas/CodePrinter.fs#L5138-L5141

We would need to pipe `ifElse isInline (!- "inline ") sepNone` with `tokN estimatedRange INLINE`.
Where `estimatedRange` is a range where we will find the `inline` token in.
You could construct this range based on neighbouring nodes.

Are you interesting in submitting a PR for this one?

**Comment by TheAngryByrd** [comment]
Thanks for pointing out where I need to look!

**Comment by TheAngryByrd** [comment]
Closed as this is a weird edge case and effort should be spent fixing the main problem in f# core.

**Comment by nojaf** [comment]
Fix available in [v5.2.0-beta-001](https://www.nuget.org/packages/fantomas/5.2.0-beta-001)
